### DISCUSSION TITLE
What is the purpose of `registerEnumType` if enums are already working without it ?

### DISCUSSION THREAD

**Discussion by Bradzer** [MichalLytek/type-graphql#1336] [Q&A] [answered]
The docs recommend using `registerEnumType` to work with enums.
However, even without using `registerEnumType` my enums are already working as expected.

So I wonder what is the purpose of `registerEnumType`. 

Or was it working because apollo somehow _resolves_ things behind the scenes ?

**Comment by MichalLytek** [comment, marked as answer]
Without `registerEnumType`, the generated GQL type would be String or Float (depending on the enum type), rather than the ENUM we are aiming for.

https://typegraphql.com/docs/enums.html#using-enum

**Reply by Bradzer** [reply]
Thank you for your quick response @MichalLytek.

Here is the schema I got without using `registerEnumType`. My enum `AccessStatus` does not show as `String` or `Float` like I would expect when not using `registerEnumType`.
```ts
GraphQLSchema {
  __validationErrors: [],
  description: undefined,
  extensions: undefined,
  astNode: undefined,
  extensionASTNodes: [],
  _queryType: Query,
  _mutationType: Mutation,
  _subscriptionType: undefined,
  _directives: [ @include, @skip, @deprecated, @specifiedBy ],
  _typeMap: [Object: null prototype] {
    Query: Query,
    UserResponse: UserResponse,
    User: User,
    ID: ID,
    String: String,
    Boolean: Boolean,
    Access: Access,
    AccessStatus: AccessStatus,
    Space: Space,
    Timestamp: Timestamp,
    Float: Float,
    Mutation: Mutation,
    SpaceResponse: SpaceResponse,
    RegisterArgs: RegisterArgs,
    LoginArgs: LoginArgs,
    __Schema: __Schema,
    __Type: __Type,
    __TypeKind: __TypeKind,
    __Field: __Field,
    __InputValue: __InputValue,
    __EnumValue: __EnumValue,
    __Directive: __Directive,
    __DirectiveLocation: __DirectiveLocation
  },
  _subTypeMap: [Object: null prototype] {},
  _implementationsMap: [Object: null prototype] {}
}
```

**Reply by Bradzer** [reply]
@MichalLytek 
Oh it seems like the `registerEnumType` was well hidden in the code base 😅:
```ts
registerEnumType(AccessStatus, {
  name: "AccessStatus",
});
```
